libs/@local/hash-backend-utils/src/pg-tables.ts

```diff
@@ -1,75 +1,106 @@
 /**
- * This module contains type definitions for tables in the Postgres database. Column
- * names are converted from snake_case to camelCase for consistency.
+ * This module contains type definitions for some of the tables in the Postgres database used for realtime updates.
  */
+import { SupportedRealtimeTable } from "./realtime";
 import { Wal2JsonMsg } from "./wal2json";
 
-export class Entity {
-  constructor(
-    /**
-     * @todo Update table definition when provenance info is added for updates
-     *   https://app.asana.com/0/1202805690238892/1202848989198291/f
-     */
-    public entityId: string,
-    public version: string,
-    public entityTypeVersionId: string,
-    public properties: Record<string, unknown>,
-    public createdBy: string,
-  ) {}
+export type PgEntityEdition = {
+  entity_edition_id: string;
+  properties: Record<string, unknown>;
+  left_to_right_order?: number;
+  right_to_left_order?: number;
+  record_created_by_id: string;
+  archived: boolean;
+};
 
-  private static parseFromRow(row: Record<string, unknown>): Entity {
-    return {
-      entityId: row.entity_id as string,
-      version: row.version as string,
-      entityTypeVersionId: row.entity_type_version_id as string,
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      properties:
-        typeof row.properties === "string"
-          ? JSON.parse(row.properties)
-          : row.properties,
-      createdBy: row.created_by as string,
-    };
+export const entityEditionFromWalJsonMsg = (
+  msg: Wal2JsonMsg<SupportedRealtimeTable>,
+): PgEntityEdition => {
+  if (msg.table !== "entity_editions") {
+    throw new Error(
+      `invalid table "${msg.table}" for 'entity_edition' parsing`,
+    );
   }
+  const obj = Object.fromEntries(
+    msg.columns.map(({ name, value }) => [name, value]),
+  );
 
-  static parseWal2JsonMsg(msg: Wal2JsonMsg): Entity {
-    if (msg.table !== "entities") {
-      throw new Error(`invalid table "${msg.table}" for an 'entities' update`);
-    }
-    const obj = Object.fromEntries(
-      msg.columns.map(({ name, value }) => [name, value]),
+  return {
+    entity_edition_id: obj.entity_edition_id as string,
+    properties: obj.properties as Record<string, unknown>,
+    left_to_right_order: obj.left_to_right_order as number,
+    right_to_left_order: obj.right_to_left_order as number,
+    record_created_by_id: obj.record_created_by_id as string,
+    archived: obj.archived as boolean,
+  };
+};
+
+export type PgEntityTemporalMetadata = {
+  owned_by_id: string;
+  entity_uuid: Record<string, unknown>;
+  entity_edition_id: string;
+  decision_time: string;
```

Comment:
We do need that to some degree. The consumers of the stream need to be aware of the decision time to know whether or not an actual newer version has been created. This type of notification service is really difficult because of our bitemporality, but we assume that the change stream is just a view into the table insertions, and services must find/use their own understanding of their entities to act accordingly